lib/modules/platform/gitlab/readme.md

```diff
@@ -2,39 +2,46 @@
 
 ## Authentication
 
-First, [create a Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html) for the bot account.
+You can authenticate Renovate to GitLab, with a Personal Access Token, or Group Access Token.
 
-If you are using a Group access token, the token must have Developer role or higher permissions in order to create issues and merge requests.
-The token must have Maintainer permissions in order to perform Automerge.
+To start, create either:
 
-For real runs, give the PAT these scopes:
+- a [Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html) for the bot account
+- or a [Group Access Token](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html#bot-users-for-groups) for the bot account
+
+The bot account must have at least the Developer role in order to [create issues and merge requests](https://docs.gitlab.com/ee/user/permissions.html#project-members-permissions).
+If you are using automerge, the bot account must have the appropriate ["Allowed to merge" permission on the protected branch](https://docs.gitlab.com/ee/user/project/protected_branches.html#require-everyone-to-submit-merge-requests-for-a-protected-branch) of your projects.
+If merging is restricted to Maintainers, the bot account must have the Maintainer role.
+
+If you are using a group access token, to keep using the same GitLab-generated bot user you must [rotate/refresh the Group Access Token](https://docs.gitlab.com/ee/api/group_access_tokens.html#rotate-a-group-access-token) _before_ the token's expiry date.
+
+We refer to personal access tokens and group access tokens as _access tokens_ in the instructions that follow.
+
+For real runs, give the access token these scopes:
 
-- `read_user`
 - `api`
 - `write_repository`
 - `read_registry` (only if Renovate needs to access the [GitLab Container registry](https://docs.gitlab.com/ee/user/packages/container_registry/))
 
-For dry runs, give the PAT these scopes:
+For dry runs, give the access token these scopes:
 
-- `read_user`
 - `read_api`
 - `read_repository`
-- `write_repository` (when using autodiscover)
```

Comment:
why this is removed? autodiscover will check for write access and filter repo without this permission!

https://github.com/renovatebot/renovate/blob/f21efd3c31c44367181d69f38a6e29674a653afd/lib/modules/platform/gitlab/index.ts#L168

Reply:
I tested autodiscover with `read_api` and `read_repository` and it was able to discover repositories.

`min_access_level = 30` refers to the Developer role (https://docs.gitlab.com/ee/api/members.html#roles) which is the minimum requirement for the bot account.

Reply:
so the user or group has write access, but the token is limited to read-only? strange behavior 🤔

Reply:
The way I understand it is that the role gives you [access/permissions to certain features](https://docs.gitlab.com/ee/user/permissions.html#project-members-permissions). If the token has the `read_api` scope then anything that the role has access to can be read.